pkg/codec/modifier_base.go

```diff
@@ -422,53 +422,21 @@ func valueForPath(from reflect.Value, itemType string) (any, error) {
 		}
 
 		return valueForPath(field, tail)
-	default:
-		return nil, fmt.Errorf("%w: cannot extract a field from kind %s", types.ErrInvalidType, from.Kind())
-	}
-}
-
-func applyValueForPath(vInto, vField reflect.Value, itemType string) error {
-	switch vInto.Kind() {
-	case reflect.Pointer:
-		if !vInto.Elem().IsValid() {
-			into := reflect.New(vInto.Type().Elem())
-
-			vInto.Set(into)
-		}
-
-		err := applyValueForPath(vInto.Elem(), vField, itemType)
-		if err != nil {
-			return err
-		}
-
-		return nil
-	case reflect.Array, reflect.Slice:
-		return fmt.Errorf("%w: cannot set a field from an array or slice", types.ErrInvalidType)
-	case reflect.Struct:
+	case reflect.Map:
```

Comment:
Glad to see this added! Should we add map support to the other helpers as well?

- SetValueFromPath
- ApplyValue
- RetypeToOffChain

Reply:
Done. I exported the helpers and added map support to both